Mathlib/Tactic/Widget/Calc.lean

```diff
@@ -124,19 +126,30 @@ def CalcPanel : Component CalcParams :=
   mk_rpc_widget% CalcPanel.rpc
 
 namespace Lean.Elab.Tactic
-open Meta
+
+open Lean Meta Tactic TryThis in
+/-- Create a `calc` proof. -/
+elab stx:"calc?" : tactic => withMainContext do
+  let some calcRange := (← getFileMap).rangeOfStx? stx | throwError "calc? failed"
+  let indent := calcRange.start.character + 2
+  let spc := String.replicate indent ' '
+  let goalType ← getMainTarget
```

Comment:
```suggestion
  let goalType ← whnfR (← getMainTarget)
```
to remove stuff like `mdata` from the `Expr`.  You can test that something like this is helpful with
```lean
example : 1 = 1 := by
  have := 0
  calc?
```

Reply:
Good catch, thanks!